obspy/core/tests/test_code_formatting.py

```diff
@@ -22,16 +47,32 @@ class CodeFormattingTestCase(unittest.TestCase):
     """
     @unittest.skipIf(CLEAN_VERSION_NUMBER,
                      "No code formatting tests for release builds")
+    @unittest.skipIf(not HAS_FLAKE8_AT_LEAST_VERSION_3,
+                     "Formatting tests require at least flake8 3.0.")
     @unittest.skipIf('OBSPY_NO_FLAKE8' in os.environ, 'flake8 check disabled')
     def test_flake8(self):
         """
         Test codebase for compliance with the flake8 tool.
         """
-        report, message = check_flake8()
-        file_count = report.counters["files"]
-        error_count = report.get_count()
-        self.assertGreater(file_count, 10)
-        self.assertEqual(error_count, 0, "\n" + message.decode())
+        # Import the legacy API as flake8 3.0 currently has not official
+        # public API - this has to be changed at some point.
+        from flake8.api import legacy as flake8
+        style_guide = flake8.get_style_guide(ignore=FLAKE8_IGNORE_CODES)
+
+        untracked_files = get_untracked_files_from_git() or []
+        files = []
+        for filename in get_all_py_files():
+            if filename in untracked_files:
+                continue
+            for pattern in FLAKE8_EXCLUDE_FILES:
+                if fnmatch.fnmatch(filename, pattern):
+                    break
+            else:
+                files.append(filename)
+        report = style_guide.check_files(files)
+
+        # Make sure no error occured.
+        assert report.total_errors == 0
```

Comment:
@krischer @mbyt since merging this PR we do not get any information on flake8 fails anymore, due to the plain "assert" without failure message, see e.g. https://travis-ci.org/obspy/obspy/jobs/202193359#L1159

![screenshot from 2017-02-16 13 41 17](https://cloud.githubusercontent.com/assets/1842780/23021828/31cc3bc8-f44e-11e6-8233-ba74b615883b.png)

Any reason for this?